### PR TITLE
update to support newer parallel computing toolbox syntax

### LINKED ISSUE #1009: matlabpool / parpool Dependency Check
This is line 494 of sampledFiniteTimeVerification.m:
```if (matlabpool('size')==0) matlabpool; end```
The problem is that, reportedly, ```matlabpool``` [was replaced](http://stackoverflow.com/questions/19131126/how-do-i-create-a-parallel-loop) by ```parpool``` in R2013b of the Parallel Computing Toolbox.  I stumbled into this because I am using the R2015a.  The quick fix would be:
```matlab
try
  if (matlabpool('size')==0) matlabpool; end
catch
  if isempty(gcp('nocreate')) parpool; end
end
```
However, I know that ```matlabpool``` is called in other parts of Drake.  Perhaps the solution you prefer is something like expanding and using the ```checkDependency()``` function to select the command to use according the the user's version, since this is more like the customary way of doing things I'm used to see in Drake.  Let me know and if the fix you have in mind is "communicable" I'll be happy to do it myself.

### PR DESCRIPTION
**Pull request by RussTedrake**
resolves #1009

### CODE REVIEW

File: examples/LinearGaussianExample.m [hunk 1]
```diff
@@ -40,6 +40,7 @@
  methods
    function prob=monteCarlo(obj,S,tspan,x0)
      N=200;
+     checkDependency('distcomp');   % note: if you don't have distcomp, then
```
Comment (avalenzu) [inline examples/LinearGaussianExample.m]: Should there be more to this comment?

Comment (RussTedrake) [inline examples/LinearGaussianExample.m]: yup.  fixed.

File: systems/@PolynomialSystem/sampledFiniteTimeVerification.m [hunk 2]
```diff
@@ -36,6 +36,7 @@
 % Implements the algorithm described in http://arxiv.org/pdf/1010.3013v1
 
 checkDependency('sedumi');
+checkDependency('distcomp'); % note: to operate without this, simply change the parfor to for in the loop below
```
Comment (avalenzu) [inline systems/@PolynomialSystem/sampledFiniteTimeVerification.m]: This shouldn't actually be necessary. I can't find it in the docs, but my recollection (supported by comments online) is that parfor will still run, albeit serially, on systems without PCT.

**Comment by avalenzu** [comment]
Looks fine to me. We should check if we really need to call `checkDependency('distcomp')` before using `parfor`.

File: systems/@PolynomialSystem/sampledFiniteTimeVerification.m (hunk 2 above)
Comment (RussTedrake) [inline systems/@PolynomialSystem/sampledFiniteTimeVerification.m]: oh.  that would be much better.  will see if I can try it out.

**Comment by RussTedrake** [comment]
it looks like you're correct:
http://www.mathworks.com/matlabcentral/answers/13507-does-parfor-make-a-difference-without-parallel-computing-toolbox
(i found it difficult to properly test it on my machine though).  that's much better.  will update.

**Pull request merged**